### PR TITLE
Update docs with http optout info

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by hodgestar** [comment]
I wonder whether we should explain that `to_addr_type` and `from_addr_type` default to `msisdn` and list the other possible values?

**Comment by hodgestar** [comment]
Otherwise looks good. :)

**Comment by rudigiesler** [comment]
We could do that. I didn't see anywhere in the docs where it explains what all the fields are and what each field is though, but I'll add it for the address type so long.

**Comment by hodgestar** [comment]
Tx.

File: docs/http_api.rst [hunk 1]
```diff
@@ -95,6 +104,10 @@ The response to the PUT request is the complete Vumi Go user message
 and includes the generated Vumi ``message_id`` which should be stored
 if you wish to be able to associate events with the message later.
 
+If a message is sent to a recipient that has opted out, the response will be an
+HTTP 400 error, with the body detailing that the recipient has opted out.
+Messages sent as a reply will still go through to an opted out recipient.
```
Comment (justinvdm) [inline docs/http_api.rst]: I think we should maybe mention how the error looks, like mention that `success` will be `false` and that the `reason` field will detail that the recipient has opted out.

Comment (justinvdm) [inline docs/http_api.rst]: Separate point -- should we mention that this behaviour can be overridden by setting the account `disable_optouts` flag to `true`?

**Comment by justinvdm** [comment]
Left two minor comments, looks good.

**Comment by hodgestar** [comment]
:+1: from me, although someone should review the commit I added.

**Comment by justinvdm** [comment]
:+1: :cake: for keeping the docs up to date

**Pull request merged**